guidance/library/_json.py

```diff
@@ -718,10 +703,182 @@ def anyOf(
         lm,
         *,
         anyof_list: Sequence[JSONSchema],
+        base_uri: str,
     ):
-        options = [self.json(json_schema=item) for item in anyof_list]
+        options = [self.json(json_schema=item, base_uri=base_uri) for item in anyof_list]
         return lm + select(options)
 
+    @guidance(stateless=True)
+    def oneOf(
+        self,
+        lm,
+        *,
+        oneof_list: Sequence[JSONSchema],
+        base_uri: str,
+    ):
+        if len(oneof_list) == 1:
+            return lm + self.json(json_schema=oneof_list[0], base_uri=base_uri)
+        warnings.warn("oneOf not fully supported, falling back to anyOf. This may cause validation errors in some cases.")
+        return lm + self.anyOf(anyof_list=oneof_list, base_uri=base_uri)
+
+    @guidance(stateless=True)
+    def allOf(
+        self,
+        lm,
+        *,
+        parent_schema: JSONSchema,
+        base_uri: str,
+    ):
+        type = set(JSONType)
+        properties: defaultdict[str, list[JSONSchema]] = defaultdict(list)
+        required: set[str] = set()
+        additional_properties_list: list[JSONSchema] = []
+        items_list: list[JSONSchema] = []
+        other_data: dict[str, Any] = {}
+        enum: Optional[list[Any]] = None
+        const: Union[Unset, Any] = _unset
+
+        def handle_keyword(key: str, value: Any, base_uri: str):
+            nonlocal type
```

Comment:
I have learned a new Python keyword today.....

Reply:
Super handy for modifying closure variables :)

Reply:
Sort of 'closed but not really' ? :-)